File: Solutions/1. Two Sum/sol.py
#O(N)
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        d = {} 
        for i in range(len(nums)):
            num2 = target - nums[i]
            if num2 in d:
                return [i , d[num2]]
            
            d[nums[i]] = i 

File: Solutions/1. Two Sum/sol.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int,int> m ;
        vector<int> ans ;
        for(int i = 0 ; i < nums.size() ; i++)
        {
            int x = target - nums[i] ;
            if(m.count(x))
            {
                ans.push_back(i) ;
                ans.push_back(m[x]) ;
                break ;
            }
            m[nums[i]] = i ;
        }
        return ans ;
    }
};